hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/SolvencyPreCheck.java

```diff
@@ -145,6 +145,9 @@ private long estimateAdditionalCosts(@NonNull final TransactionInfo txInfo, @Non
         return switch (txInfo.functionality()) {
             case CRYPTO_CREATE -> txInfo.txBody().cryptoCreateAccountOrThrow().initialBalance();
             case CRYPTO_TRANSFER -> {
+                if (!txInfo.txBody().cryptoTransferOrThrow().hasTransfers()) {
+                    yield 0L;
```

Comment:
Why do we need this check ? Wouldn't we catch NPE is there are no transfers and fail with `InsufficientBalanceException(INVALID_TRANSACTION_BODY, totalFees)` ?

Reply:
Ah I see. Makes sense. Thanks ! Probably there is a check to see if both `TransferList` and `TokenTransferList` is null. Then it should throw `INVALID_TRANSACTION_BODY`